docs/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,219 @@
+  + [Initialization](#initialization)
+  + [Fetching Products and Collections](#fetching-products-and-collections)
+  + [Carts/Checkouts](#cartscheckouts)
+
+## Installation
+
+**With Yarn:**
+
+```bash
+yarn add shopify-buy@alpha
+```
+**With NPM:**
+
+Remove the old version, then
+```bash
+npm install shopify-buy@alpha
+```
+
+## Updated Functions and Classes
+
+### Initialization
+
+Rather than using a static function to create the client (`ShopifyBuy.buildClient()`), v1 exposes the `Client`
+(previously `ShopClient`) and a `Config` class for the `Client` directly. An instance of `Client` can be created like so:
+```js
+import Client, {Config} from 'shopify-buy';
+
+const config = new Config({
+  domain: 'your-shop-name.myshopify.com',
+  storefrontAccessToken: 'your-storefront-access-token'
+});
+
+const client = new Client(config);
+```
+
+### Fetching Products and Collections
+
+The functions for fetching products and collections are mostly the same. Major differences are:
+
+1.  The v1 functions take in a `Storefront ID` for fetching a product or collection by ID. A `Storefront ID` can be found under
+    the `API details` section on product and collection pages in the Shopify admin. 
+    `API details` will only be displayed if you have a private app with Storefront API access. For more details see the 
+    [Getting Started Guide](https://help.shopify.com/api/storefront-api/getting-started#authentication) for the Storefront API.
+
+2.  The functions now take an additional optional `query` argument that specifies the fields to query on the resource.
+    ```js
+    const query = productNodeQuery(['title', ['images', imageConnectionQuery()]];
+
+    client.fetchProduct('Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0Lzk4OTUzMTMwMjc=', query).then((product) => {
+      console.log(product); // Product with only title and images
+    });
+    ```
+    Fields and queries can be nested like so:
+    ```js
+    // Product query with title and variants, where variants have price and title
+    const query = productNodeQuery(['title', ['variants', variantConnectionQuery(['price', 'title'])]]);
+    ```
+    If the `query` argument isn't supplied, a default list of arguments will be used.
+    See the [API reference](https://github.com/Shopify/js-buy-sdk/blob/v1.0alpha/docs/API_REFERENCE.md#Client.Queries) for more details.
+
+3.  `fetchCollection()` and `fetchAllCollections()` no longer return the products in the collections by default.
+    To fetch products on the collections, use `fetchCollectionWithProducts()` and `fetchAllCollectionsWithProducts()`,
+    or specify products in the query function parameter.
+    ```js
+    const collectionId = 'Z2lkOi8vc2hvcGlmeS9Db2xsZWN0aW9uLzI1NzY5NzczMQ=='
+    
+    // Use the built-in function
+    client.fetchCollectionWithProducts(collectionId).then((collection) => {
+      console.log(collection); // Collection with all default fields and products with all default fields.
```

Comment:
Show where the products live. It seems obvious to us that `collection.products` is an array, but externally sometimes this stuff isn't obvious.